http/http.go

```diff
@@ -0,0 +1,86 @@
+package http
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// Response is the type that all the HTTP requests are going to return.
+type Response struct {
+	Code    int
+	Body    interface{}
+	Headers http.Header
+}
+
+//Request is the
+type Request struct {
+	URL     string
+	Headers map[string]string
+	Body    []byte
+}
+
+// Get executes a GET request against the specified URL.
+func Get(r Request, out interface{}) (*Response, error) {
+	return do("GET", r, out)
+}
+
+// Post executes a POST request against the specified URL.
+func Post(r Request, out interface{}) (*Response, error) {
+	return do("POST", r, out)
+}
+
+// Patch executes a PATCH request against the specified URL.
+func Patch(r Request, out interface{}) (*Response, error) {
+	return do("PATCH", r, out)
+}
+
+// Put executes a PUT request against the specified URL.
+func Put(r Request, out interface{}) (*Response, error) {
+	return do("PUT", r, out)
+}
+
+func do(method string, request Request, out interface{}) (*Response, error) {
+
+	// Build the request
+	req, err := http.NewRequest(method, request.URL, bytes.NewBuffer(request.Body))
+	if err != nil {
+		return nil, fmt.Errorf("Error building the %s request for %s: %v", method, request.URL, err)
+	}
+
+	// Fill the request with the headers
+	for k, v := range request.Headers {
+		req.Header.Set(k, v)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
```

Comment:
I remember reading a while ago that it is better to not use the default client of the http package. Not sure if that is the case anymore, but may want to change it?

```Go
client := &http.Client{}
// Not sure what a best 'default' timeout would be, 10 seconds here
client.Timeout = time.Second * time.Duration(10)
resp, err := client.Do(req)
```

Reply:
Yeah, I'll change it now: https://medium.com/@nate510/don-t-use-go-s-default-http-client-4804cb19f779

>TL;DR: Go’s http package doesn’t specify request timeouts by default, allowing services to hijack your goroutines. Always specify a custom http.Client when connecting to outside services.